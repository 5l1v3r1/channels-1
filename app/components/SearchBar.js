// @flow
import React from 'react';
import { IconSearch, IconCancel } from './Icons';
import styles from './SearchBar.scss';

const SearchBar = ({
  disabled,
  placeholder
}: {
  disabled: boolean,
  placeholder: string
}) => (
  <form className={styles.searchBar}>
    {disabled && <IconSearch styles={styles} />}
    <input
      className={styles.searchInput}
      disabled={!disabled}
      placeholder={placeholder}
    />
    <IconCancel styles={styles} />
  </form>
);

export default SearchBar;
