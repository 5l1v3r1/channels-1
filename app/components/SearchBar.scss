@import '../app.global.scss';

.searchBar {
  display: flex;
  align-items: center;
  height: 100%;
  padding: 0 1rem;
  background-color: $black;
}

.iconSearch {
  height: 24px;
  fill: $gray;
}

.searchInput {
  display: flex;
  flex-grow: 1;
  line-height: 24px;
  padding: 0 0.5rem;
  border: none;
  color: $white;
  background-color: transparent;

  &:focus {
    outline: 0;
  }
}

.resetBtn {
  display: flex;
  justify-content: center;
  width: 24px;
  height: 24px;
  padding: 0;
  border: none;
  background-color: transparent;
  cursor: default;

  &:focus {
    outline: 0;
  }
}

.iconCancel {
  width: 14px;
  fill: $gray;
}
