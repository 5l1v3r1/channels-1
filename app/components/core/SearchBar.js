// @flow
import React from 'react';
import { IconSearch, IconCancel } from './Icons';
import styles from './SearchBar.scss';

const SearchBar = ({
  disabled,
  placeholder,
  value,
  handleChange,
  handleReset
}: {
  disabled: boolean,
  placeholder: string,
  value: string,
  handleChange: (event: SyntheticInputEvent<HTMLInputElement>) => void,
  handleReset: () => void
}) => (
  <div className={styles.searchBar}>
    <IconSearch className={disabled ? [styles.iconSearch, styles.isDisabled].join(' ') : styles.iconSearch} />
    <input
      className={styles.searchInput}
      disabled={disabled}
      placeholder={placeholder}
      value={value}
      onChange={handleChange}
    />
    {value &&
      <button className={styles.resetBtn} type="reset" onClick={handleReset}>
        <IconCancel className={styles.iconCancel} />
      </button>
    }
  </div>
);

export default SearchBar;
