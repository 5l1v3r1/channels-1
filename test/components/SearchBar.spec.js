import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from '../../app/components/SearchBar';

function setup() {
  const props = {
    disabled: false,
    placeholder: 'Search DevTips',
    value: '',
    handleChange: jest.fn(),
    handleReset: jest.fn()
  };

  const enzymeWrapper = shallow(<SearchBar {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('SearchBar', () => {
  describe('when disabled', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      enzymeWrapper.setProps({
        disabled: true,
        placeholder: 'The Mac App For YouTube Channels.'
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when enabled', () => {
    it('should render self with no value', () => {
      const { enzymeWrapper } = setup();
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });

    it('should render self with value', () => {
      const { enzymeWrapper } = setup();
      enzymeWrapper.setProps({
        value: 'Sass'
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  it('should call handleChange', () => {
    const { props, enzymeWrapper } = setup();
    const searchInput = enzymeWrapper.find('.searchInput');

    searchInput.simulate('change');

    expect(props.handleChange).toBeCalled();
  });

  it('should call handleReset', () => {
    const { props, enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      value: 'Sass'
    });
    const resetBtn = enzymeWrapper.find('.resetBtn');

    resetBtn.simulate('click');

    expect(props.handleReset).toBeCalled();
  });
});
