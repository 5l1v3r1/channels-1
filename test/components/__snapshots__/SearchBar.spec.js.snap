// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar when disabled should render self 1`] = `
<div
  className="searchBar"
>
  <input
    className="searchInput"
    disabled={true}
    onChange={[Function]}
    placeholder="The Mac App For YouTube Channels."
    value=""
  />
</div>
`;

exports[`SearchBar when enabled should render self with no value 1`] = `
<div
  className="searchBar"
>
  <Component
    className="iconSearch"
  />
  <input
    className="searchInput"
    disabled={false}
    onChange={[Function]}
    placeholder="Search DevTips"
    value=""
  />
</div>
`;

exports[`SearchBar when enabled should render self with value 1`] = `
<div
  className="searchBar"
>
  <Component
    className="iconSearch"
  />
  <input
    className="searchInput"
    disabled={false}
    onChange={[Function]}
    placeholder="Search DevTips"
    value="Sass"
  />
  <button
    className="resetBtn"
    onClick={[Function]}
    type="reset"
  >
    <Component
      className="iconCancel"
    />
  </button>
</div>
`;
